### PR TITLE
test: replace cli with go-ceph for snapshot namespace PR

### PR DESCRIPTION
**Pull request by Madhu-1**
This PR is to test changes of https://github.com/ceph/ceph-csi/pull/1446 and to avoid panic in E2E tests

### CODE REVIEW

**Comment by Madhu-1** [comment]
This is just a testing PR
cc @agarwal-mudit

**Comment by agarwal-mudit** [comment]
Thanks @Madhu-1

File: internal/rbd/rbd_util.go [hunk 1]
```diff
@@ -426,9 +426,24 @@ func flattenClonedRbdImages(ctx context.Context, snaps []snapshotInfo, pool, mon
 		util.ErrorLog(ctx, "failed to open connection %s; err %v", rv, err)
 		return err
 	}
-	for _, s := range snaps {
-		if s.Namespace.Type == "trash" {
-			rv.RbdImageName = s.Namespace.OriginalName
+
+	for _, snapInfo := range snaps {
+		// check if the snapshot belongs to trash namespace.
+		isTrash, err := rv.isTrashSnap(snapInfo.Id)
```
Comment (Madhu-1) [inline internal/rbd/rbd_util.go:432]: @agarwal-mudit I think this won't be valid as the rbd image name will be updated at line 446 isn't it?

Comment (agarwal-mudit) [reply internal/rbd/rbd_util.go:432]: Nice catch @Madhu-1, Yeah that should be an issue here. Let me think of a solution. Thanks for the help.

**Pull request closed without merging**